lib/active_record/connection_adapters/makara_abstract_adapter.rb

```diff
@@ -212,7 +212,12 @@ def initialize(proxy)
           @proxy = proxy
           @owner = nil
           @pool = nil
-          @schema_cache = ActiveRecord::ConnectionAdapters::SchemaCache.new @proxy
+          if ActiveRecord.version >= Gem::Version.new('7.1.0')
+            schema_reflection = ActiveRecord::ConnectionAdapters::SchemaReflection.new(nil).load!(@proxy)
```

Comment:
This still not going to work because you are passing `nil` to `ActiveRecord::ConnectionAdapters::SchemaReflection.new`

The right approach would be passing the schema_cache_path to `ActiveRecord::ConnectionAdapters::SchemaReflection.new` in order to fix it